### PR TITLE
docs(http-security): refresh stale inventories and close navigation gaps

### PR DESCRIPTION
**Pull request by OliverWolffGIP**
## Summary

Refreshes cui-http's documentation tree so it matches the current codebase and closes its own navigation gaps — a scope-corrective accuracy pass, not a restructure. Corrects stale test/pipeline/validation-stage counts and their cross-referenced traceability rows, links two orphaned documents into the relevant readmes, adds two missing security README index entries, records two missing implementation-status notes, refreshes `CLAUDE.md`'s architecture section and the pipeline-selection decision matrix, and fixes several smaller accuracy items (runtime-module count, a misdirected cross-reference, RFC 8941/7230 citations, stale tree-root prefixes, an overdue review date, dangling Javadoc references).

## Changes

- `CLAUDE.md` — refreshed architecture section to list all 5 concrete pipelines and all exported client packages (`de.cuioss.http.forwarded`, `client.adapter`, `client.converter`, `client.result`)
- `README.adoc` — navigation and accuracy corrections
- `doc/client-handlers-readme.adoc` — accuracy corrections
- `doc/http-security/README.adoc` — added missing Document Index entries, linked orphaned documents
- `doc/http-security/functional-requirements.adoc` — added missing implementation-status notes (HTTP-1 body validation throws; HTTP-14 cookie per-type configs not yet implemented)
- `doc/http-security/specification/compliance-traceability.adoc` — corrected stale counts and traceability rows
- `doc/http-security/specification/pipeline-architecture-standards.adoc` — corrected pipeline counts, extended the pipeline-selection decision matrix beyond `URLPathValidationPipeline` alone
- `doc/http-security/specification/specification.adoc` — corrected stale counts and cross-references
- `doc/http-security/specification/testing.adoc` — corrected stale test counts
- `doc/test-framework-structure.adoc` — corrected stale test-tree counts and prefixes
- `doc/test-generators-readme.adoc` — accuracy corrections

## Test Plan

- [ ] Verification command passed (`python3 .plan/execute-script.py plan-marshall:build-maven:maven run --command-args "verify -Ppre-commit"`)
- [ ] Manual testing completed (if applicable)

## Related Issues

None.

---
Generated by plan-finalize skill

## Intent

The cui-http documentation tree had drifted behind the code: stale test/pipeline/validation-stage counts, two orphaned documents with zero inbound references, missing implementation-status notes, and an architecture section in CLAUDE.md that named 3 pipelines against the 5 that actually exist.

The fix is a scope-corrective accuracy pass, not a restructure. Every count claim was re-verified against HEAD by name (not line number) since a concurrent test-tree workstream may still be landing changes. Where a document hand-maintained an exhaustive inventory the tree already owns, the inventory is replaced by a pointer to the source of truth rather than re-typed with today's numbers. Where a count carries independent meaning and must stay in prose, it is corrected together with its paired traceability-table row in the same edit, per the paired-claim discipline from a prior landing lesson.

This change does not touch cui-http-core/src/main/java/de/cuioss/http/security/** or any test class — those trees are owned by sibling workstreams and are consulted here only as the source of truth, never edited.

## Summary by CodeRabbit

* **Documentation**
  * Expanded HTTP security documentation with validation pipeline selection guidance.
  * Documented supported validation types, body-validation limitations, cookie-prefix validation, and module requirements.
  * Added references for HTTP result handling and the `HTTP-nnn` message catalogue.
  * Updated compliance, standards, RFC, and cross-reference guidance.
  * Reorganized testing documentation into package-level navigation maps.
  * Added configuration and compliance-traceability documentation to the security index.
  * Added architectural decision records for package-level inventories and named documentation anchors.

### CODE REVIEW

**Comment by OliverWolffGIP** [comment]
@coderabbitai review

File: doc/http-security/functional-requirements.adoc [hunk 1]
```diff
@@ -30,13 +30,20 @@ These requirements are derived from:
 * The system must provide separate validators for each component type
 * Reference: xref:specification/specification.adoc#_core_components[Core Components]
 
+NOTE: *Body validation is not provided.* `PipelineFactory.createPipeline(ValidationType.BODY, ...)`
+throws `IllegalArgumentException` with the message "BODY validation pipeline has been removed.
+HTTP body content validation should be handled at application layer." Request-body parameters are
+therefore an application-layer responsibility, not a capability of this library. See
+xref:#HTTP-14[HTTP-14] for the per-validation-type status.
+
```
[the remark this answers is not in the document]
Comment (OliverWolffGIP) [reply doc/http-security/functional-requirements.adoc]: Will be addressed by TASK-7; see follow-up commit on this branch

File: doc/http-security/README.adoc [hunk 2]
```diff
@@ -77,14 +82,18 @@ According to RFC 3986, valid URL path components must adhere to:
...
+=== Query-Parameter Name Validation
 
-* Must start with alphanumeric character
-* Can contain: `A-Z`, `a-z`, `0-9`, `-`, `_`
-* Case-insensitive in headers, case-sensitive in query parameters
-* Maximum length considerations for security
+Query-parameter names are *not* governed by RFC 7230 — they are validated by
+`URLParameterNameValidationPipeline` under RFC 3986 query-component rules, plus rejection of
+delimiters that appear only after decoding. Parameter names are case-sensitive, and length
+limits are enforced by `LengthValidationStage` as the pipeline's first stage.
```
[the remark this answers is not in the document]
Comment (OliverWolffGIP) [reply doc/http-security/README.adoc]: Will be addressed by TASK-8; see follow-up commit on this branch

File: doc/http-security/specification/compliance-traceability.adoc [hunk 3]
```diff
@@ -687,8 +686,8 @@ audit *logging* is a caller responsibility, not provided by the validation core.
 
 ==== Test Framework
 
-* 14 Databases (10 attack databases: OWASP Top 10, OWASP ZAP, ModSecurity CRS, Apache/IIS/Nginx CVE, Homograph, IDN, IPv6, ProtocolHandler; plus 4 legitimate-pattern databases)
-* 20+ Test Generators
+* 10 attack databases (OWASP Top 10, OWASP ZAP, ModSecurity CRS, Apache/IIS/Nginx CVE, Homograph, IDN, IPv6, ProtocolHandler) and 3 legitimate-pattern databases (edge-case valid URLs, legitimate path patterns, legitimate special characters)
+* `TypedGenerator` implementations grouped by input type — see xref:testing.adoc[Testing Framework]
```
[the remark this answers is not in the document]
Comment (OliverWolffGIP) [reply doc/http-security/specification/compliance-traceability.adoc]: Will be addressed by TASK-9; see follow-up commit on this branch

File: doc/http-security/specification/testing.adoc [hunk 4]
```diff
@@ -14,137 +14,48 @@ Testing framework for HTTP security validation with generator-based testing, att
...
+
+|`pipeline`
+|Tests for `PipelineFactory` and for each concrete validation pipeline.
+
+|`tests`
+|Attack-specific test classes — one per attack family or attack database, each driving a
+pipeline with the corresponding curated patterns.
+
+|`validation`
+|Tests for the individual validation stages and for `RequestCollectionValidator`.
```
[the remark this answers is not in the document]
Comment (OliverWolffGIP) [reply doc/http-security/specification/testing.adoc]: Accepted, not implemented here: adding CI/build validation for the hardcoded package lists in testing.adoc and test-framework-structure.adoc is new tooling work outside this plan's documentation-only scope (no deliverable covers a build/validation mechanism). Tracked as a follow-up improvement rather than blocking this PR.

**Comment by OliverWolffGIP** [comment]
## Triage dispositions

### In reply to comment_id: `PRR_kwDOPwSbp88AAAABLCLDww`

Sourcery rate-limit notice (review budget exhausted), not review feedback — no action to take; not a false positive to dismiss, simply nothing to fix.

### In reply to comment_id: `IC_kwDOPwSbp88AAAABQ8C6Ig`

PR-Agent reviewer guide reports no relevant tests, no security concerns, no major issues detected — informational summary with nothing actionable.

### In reply to comment_id: `PRR_kwDOPwSbp88AAAABLEN0tQ`

CodeRabbit review-status summary comment; its 4 actionable comments are each triaged individually as separate pr-comment findings (5b0f94, 40ede7, 038ae0, e7340e).

### In reply to comment_id: `IC_kwDOPwSbp88AAAABQ_EFQA`

CodeRabbit automated acknowledgement ('Review finished') — no content to action.

File: doc/adr/0005-Documentation_cross-references_use_named_anchors_not_positional_prose.adoc [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+perfectly. A "see below" pointing at content that now sits above is not a broken link; it is a
+correct link with wrong prose, so no validator can see it. The reader is simply sent the wrong way.
+
+The failure is asymmetric with the reordering that causes it. Whoever moves a section has no signal
+that a reference elsewhere depended on the old position, and no practical way to find one — the
+dependency is expressed in English, not in markup. So the reference decays as a side effect of an
+edit made somewhere else entirely, by someone who never saw it.
+
+Anchors invert this. A named reference either resolves or it does not, which makes the same class of
+breakage mechanically detectable by the link validation the documentation build already runs.
```
[the remark this answers is not in the document]
Comment (OliverWolffGIP) [reply doc/adr/0005-Documentation_cross-references_use_named_anchors_not_positional_prose.adoc]: Fixed in 882cc7b — you were right, and this was the plan's own defect class reproduced in the ADRs written to remove it.

Verified before correcting: pom.xml configures no AsciiDoc processing, and .github/workflows/maven.yml delegates to a reusable workflow that skips documentation-only changes. No link checker runs on this repository.

Both claims are replaced with the property that actually holds. An anchor makes a broken reference DECIDABLE by a checker, where positional prose is not decidable at all — that is the precondition for automating the check, not evidence the check exists. ADR-0005 now states plainly that the breakage stays silent until a checker is added, and names adding one as the follow-on this decision enables rather than assumes. ADR-0004 now records that a broken package pointer is currently silent too, and that the trade is a move from undetectable to merely undetected.

File: doc/adr/0005-Documentation_cross-references_use_named_anchors_not_positional_prose.adoc [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+of its reference style, which is the wrong thing to hold fixed.
+
+The alternatives all leave the reference expressed in a form no tool can check, and then spend
+human attention compensating. The decision moves the reference into a form the existing validation
+already covers.
+
+== References
+
+* `doc/http-security/specification/specification.adoc` — anchored cross-reference to the two-pass
+  pattern-matching note
```
[the remark this answers is not in the document]
Comment (OliverWolffGIP) [reply doc/adr/0005-Documentation_cross-references_use_named_anchors_not_positional_prose.adoc]: Fixed in 882cc7b, same correction as the sibling thread on this file. The claim that link validation already catches these breakages was unsupported — verified that pom.xml configures no AsciiDoc processing and the Maven workflow skips documentation-only changes — so both ADRs now state that the breakage is currently silent and describe anchors as making it decidable rather than detected.
